### PR TITLE
[wip] Filter testing architecture

### PR DESCRIPTION
**Pull request by wolovim**
### What was wrong?
"Testing event filters sucks." - @pipermerriam, issue #1635 

### How was it fixed?
Pulled in large chunks of [cthaeh](https://github.com/ethereum/cthaeh/) to load logs into an in-memory sqlite db and retrieve them when methods like `getLogs` are called. `test_filters_using_db.py` contains some new tests which leverage all that new business.

Note sure at what point this is merge-ready, so getting this up to start the discussion. Feedback of all sizes welcome.

### Todo:
- [x] Awaiting model updates or package release from @pipermerriam 
- [ ] Integration with eth-orm
- [x] Add entry to the [release notes](https://github.com/ethereum/web3.py/blob/master/newsfragments/README.md)

### CODE REVIEW

File: tests/core/filtering/test_filters_using_db.py [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+def test_create_filter_handles_none_args(session, w3, emitter):
+    event_filter = w3.eth.filter({"address": None, "fromBlock": None})
+    all_logs = event_filter.get_all_entries()
+    assert len(all_logs) == 0
+    construct_log(session, address=to_canonical_address(emitter.address))
+    new_logs = event_filter.get_new_entries()
+    assert len(new_logs) == 1
+
+
+@pytest.mark.skip("Known issue: getLogs should accept None values")
```
Comment (pipermerriam) [inline tests/core/filtering/test_filters_using_db.py:130]: Typically we'd do this by actually testing the known failure using `with pytest.raises(...)` on the offending line.

**Comment by pipermerriam** [comment]
I believe I'm at a point with the database models that I've created in `cthaeh` to pull them out and release them as a standalone package.  I don't think I'll get to it today but I think I can have it done by next week.

**Comment by marcgarreau** [comment]
@pipermerriam 👍 finally resolved the lint-pocalypse, so this will sit on the shelf until then.

**Comment by pacrob** [comment]
stale!

**Pull request closed without merging**